whole-tale/ngx-dashboard
# View logs modal

## Problem
There is no way to view logs of a failed Tale build or job

Fixes #42 
  
## Approach
* Add a log viewer modal that, given a `jobId`, will fetch the logs for that job
* Auto-refresh log data every 2 seconds, cleanup refresh interval OnDestroy

## How to Test
1. Checkout this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Launch any Tale
  * You should see the notification panel automatically display with a new notification
4. Click the "View Logs" button beside the notification
  * You should see a modal display indicating that it is fetching the logs
  * You should see log data appear once it has been fetched
  * You should notice that the log data is automatically refreshes every 2 seconds (this is easier to see this in active builds, but can check Networks tab in your browser's debug console if needed)
5. Click the Close button
  * The modal should close
